### PR TITLE
Fixes for #11044,  #11076, #11081 revoked #11038 (discuss and decline)

### PR DESCRIPTION
**Pull request by comdiv**
Summary:
1) elasticsearch shell script will be usable from Windows msys bash without changes for Unix (#11044)

2) POST /index/_analyze now will have additional ?alldata=true parameter to allow include flags, increment, payload, keyword info in returned token's JSON  (#11076)

3) PUT /index/_settings now will have ability to completly remove parameters and sub-trees from
settings with special ```"__REMOVE__"``` value  (#11081)
#11076, #11081 have corresponded tests

### CODE REVIEW

File: src/main/java/org/elasticsearch/cluster/metadata/MetaData.java [hunk 1]
```diff
@@ -642,19 +642,12 @@ public String resolveIndexRouting(@Nullable String routing, String aliasOrIndex)
...
      * @param aliasesOrIndices the aliases or indices to be resolved to concrete indices
      * @return the obtained concrete indices
-<<<<<<< HEAD
      * @throws IndexMissingException if one of the aliases or indices is missing and the provided indices options
      * don't allow such a case, or if the final result of the indices resolution is no indices and the indices options
      * don't allow such a case.
      * @throws IllegalArgumentException if one of the aliases resolve to multiple indices and the provided
      * indices options don't allow such a case.
-=======
-     * @throws IndexMissingException                 if one of the aliases or indices is missing and the provided indices options
```
Comment (comdiv) [inline src/main/java/org/elasticsearch/cluster/metadata/MetaData.java]: It was after merge and i don't know where it was produced

File: bin/elasticsearch.in.sh [hunk 2]
```diff
@@ -1,6 +1,10 @@
 #!/bin/sh
 
-ES_CLASSPATH="$ES_CLASSPATH:$ES_HOME/lib/${project.build.finalName}.jar:$ES_HOME/lib/*:$ES_HOME/lib/sigar/*"
+if  [[ $OSTYPE = *win* || $OSTYPE = msys ]]; then
+	ES_CLASSPATH="$ES_HOME\\lib\\${project.build.finalName}.jar;$ES_HOME\\lib\\*;$ES_HOME\\lib\\sigar\\*"
```
Comment (comdiv) [inline bin/elasticsearch.in.sh]: It's required ro Windows java.exe

File: bin/elasticsearch [hunk 3]
```diff
@@ -54,6 +54,18 @@
...
+	echo "NOTICE: Launch simplifyed developer config for Windows"
+	# determine elasticsearch home
+	ES_HOME=`dirname "$0"`/..
+	# make ELASTICSEARCH_HOME absolute and windows-style
+	ES_HOME=$(cd "$ES_HOME"; pwd | sed -e 's/^\///' -e 's/\//\\/g' -e 's/^./\0:/')
+	. $ES_HOME/bin/elasticsearch.in.sh
+	"$JAVA_HOME/bin/java.exe" $JAVA_OPTS $ES_JAVA_OPTS $ES_PARAMS -cp "$ES_CLASSPATH" "org.elasticsearch.bootstrap.Elasticsearch"
+	exit $?;
+fi
+
```
Comment (comdiv) [inline bin/elasticsearch]: So windows logic is completly separated from unix one and cannot be conflicted

File: src/test/java/org/elasticsearch/action/IndicesRequestTests.java [hunk 4]
```diff
@@ -167,19 +177,45 @@ public void testGetFieldMappings() {
...
+        AnalyzeResponse response =  internalCluster().clientNodeClient().admin().indices().analyze(analyzeRequest).actionGet();
+        AnalyzeResponse.AnalyzeToken token = response.getTokens().get(0);
+        assertFalse(token.isAlldata());
+        XContentBuilder builder = JsonXContent.contentBuilder();
+        response.toXContent(builder,null);
+        String str = new String(builder.bytes().array(), Charset.forName("UTF8"));
+        assertFalse(str.contains("\"flags\":"));
+        clearInterceptedActions();
+        assertSameIndices(analyzeRequest, analyzeShardAction);
+    }
```
Comment (comdiv) [inline src/test/java/org/elasticsearch/action/IndicesRequestTests.java]: Think this test is more usefull rather when before. Now it has some assertions about response while test before was just about "no errors"

File: src/main/java/org/elasticsearch/cluster/metadata/MetaData.java [hunk 5]
```diff
@@ -1393,6 +1386,47 @@ public Builder customs(ImmutableOpenMap<String, Custom> customs) {
...
+                    String key = setting.getKey();
+                    String value = setting.getValue();
+                    cleanedBuilder.put(key, value);
+                    //avoid strange behavior of Java binder that can choose Object... notation of ctor
+                }
+            }
+            final Settings cleanedTargetSettings = cleanedBuilder.build();
+            final Settings updatedSettings = settingsBuilder().put(cleanedTargetSettings).put(cleanedSourceSettings).build();
+            return updatedSettings;
+        }
```
Comment (comdiv) [inline src/main/java/org/elasticsearch/cluster/metadata/MetaData.java]: It's better to isolate setting merging process to method - that gives as far more control on it. Here we realize support for __REMOVE__ logic but we can apply any other merging logic here further.

May be it's better to move it to some SettingUtuls but while for now it is used only in MetaData.Builder i keep it there

**Comment by clintongormley** [comment]
Sorry it has taken a while to look at this.  Most issues seem to have been dealt with already.  Btw, PRs should deal with one issue at a time, rather than combining unrelated changes

thanks

**Pull request closed without merging**